pkg/endpointmanager/subscribe.go

```diff
@@ -10,12 +10,16 @@ type Subscriber interface {
 	// EndpointCreated is called at the end of endpoint creation.
 	// Implementations must not attempt write operations on the
 	// EndpointManager from this callback.
-	EndpointCreated(ep *endpoint.Endpoint)
+	// epList is the list of all endpoints that exist after the creation of
+	// the new endpoint.
+	EndpointCreated(ep *endpoint.Endpoint, epList []*endpoint.Endpoint)
 
 	// EndpointDeleted is called at the end of endpoint deletion.
 	// Implementations must not attempt write operations on the
 	// EndpointManager from this callback.
-	EndpointDeleted(ep *endpoint.Endpoint, conf endpoint.DeleteConfig)
+	// epList is the list of all endpoints that exist after the deletion of
+	// the endpoint.
+	EndpointDeleted(ep *endpoint.Endpoint, conf endpoint.DeleteConfig, epList []*endpoint.Endpoint)
```

Comment:
It seems like a bit of an abuse of this interface to bolt on `epList` when most subscribers are not going to use it. I wonder if instead of making this change to the interface and having to fetch the list of endpoints from the endpointmanager via `getEndpointsLocked`, we could build up a local cache of endpoints within the auth gc code. Upon initialization of auth gc, it can be initialized with the current list of endpoints from endpointmanager and then auth gc subscribes to `EndpointCreate` to further populate its local cache. Then that local cache can be used instead of reaching back out to the endpointmanager.